### ISSUE TITLE
Formatter is indenting more than necessary

### ISSUE THREAD

**Issue by josevalim** [elixir-lang/elixir#7005]
```elixir
  @callback init(args :: term) ::
              {
                  :ok,
                  {
                    sup_flags(),
                    [:supervisor.child_spec()]
                  }
                }
              | :ignore
```

**Comment by rcdilorenzo** [comment]
I was looking into helping with this issue, but I'm not sure what the expected indent should be. Could you provide how you would expect this snippet to be formatted?

**Comment by josevalim** [comment]
```elixir
    @callback init(args :: term) ::
                {
                  :ok,
                  {
                    sup_flags(),
                    [:supervisor.child_spec()]
                  }
                }
              | :ignore
```

**Comment by isaacsanders** [comment]
Does the formatter turn the following into the above?

```elixir
@callback init(args :: term) :: {:ok, {sup_flags(), [:supervisor.child_spec()]}} | :ignore
```

**Comment by rcdilorenzo** [comment]
I'm trying to get started on this but am not sure what section of the `Code.Formatter` or even `Inspect.Algebra` I should look in. I'll come back to it and try to work through it but thought someone might be kind enough to point me in the right direction.

**Comment by josevalim** [comment]
@rcdilorenzo I think this will be one of the bugs where it takes you 12 hours to find the fix and 2 minutes to fix it. It is definitely a bug in the Code.Formatter though. Either in the tuple or in the operator rendering.

**Comment by BenMorganIO** [comment]
Honestly, I would strongly prefer:

```elixir
@callback init(args :: term) ::
            {:ok, {sup_flags(), [:supervisor.child_spec()] } } |
            :ignore
```

So that each type is on its own line. The formatted version supplied isn't as readable.

**Comment by josevalim** [comment]
Folks, the fact the snippet above is broken does not mean your code will be formatted like that. In fact, the whole things fits on a single line, so the formatter won't even introduce a line break. The snippet above is just an example that triggered the bug post formatting. If in doubt, **try the formatter out**.

**Comment by rcdilorenzo** [comment]
Maybe I missed something but why was this issue closed? Is there another one that covers it?

**Comment by mpranjic** [comment]
@rcdilorenzo apparently fixed in commit  ca75fc6 :)

**Comment by rcdilorenzo** [comment]
Ah, sorry. I couldn't see that on mobile. :-)
